examples/demo-rollup/src/rollup.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::Context;
+use celestia::verifier::RollupParams;
+use celestia::CelestiaService;
+use demo_stf::app::{App, DefaultContext};
+use demo_stf::runtime::get_rpc_methods;
+use risc0_adapter::host::Risc0Verifier;
+use sov_db::ledger_db::LedgerDB;
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::zk::Zkvm;
+use sov_state::storage::Storage;
+use sov_stf_runner::{from_toml_path, RollupConfig, RunnerConfig, StateTransitionRunner};
+use tracing::{debug, Level};
+
+#[cfg(feature = "experimental")]
+use crate::register_rpc::register_ethereum;
+use crate::register_rpc::{register_ledger, register_sequencer};
+use crate::{get_genesis_config, initialize_ledger, ROLLUP_NAMESPACE};
+
+/// Dependencies needed to run the rollup.
+pub struct Rollup<Vm: Zkvm, DA: DaService> {
+    app: App<Vm, DA::Spec>,
+    da_service: DA,
+    ledger_db: LedgerDB,
+    runner_config: RunnerConfig,
+}
+
+/// Creates celestia based rollup.
+pub async fn new_rollup_with_celestia_da(
+    rollup_config_path: &str,
+) -> Result<Rollup<Risc0Verifier, CelestiaService>, anyhow::Error> {
+    debug!("Starting demo rollup with config {}", rollup_config_path);
+    let rollup_config: RollupConfig<celestia::DaServiceConfig> =
+        from_toml_path(rollup_config_path).context("Failed to read rollup configuration")?;
+
+    // Initializing logging
+    let subscriber = tracing_subscriber::fmt()
+        .with_max_level(Level::INFO)
+        .finish();
+    tracing::subscriber::set_global_default(subscriber)
+        .map_err(|_err| eprintln!("Unable to set global default subscriber"))
+        .expect("Cannot fail to set subscriber");
+
+    let ledger_db = initialize_ledger(&rollup_config.storage.path);
+
+    let da_service = CelestiaService::new(
+        rollup_config.da.clone(),
+        RollupParams {
+            namespace: ROLLUP_NAMESPACE,
+        },
+    )
+    .await;
+
+    let app = App::new(rollup_config.storage);
+
+    Ok(Rollup {
+        app,
+        da_service,
+        ledger_db,
+        runner_config: rollup_config.runner,
+    })
+}
+
+impl<Vm: Zkvm, DA: DaService<Error = anyhow::Error>> Rollup<Vm, DA> {
```

Comment:
How about we put the `Clone` bound here instead of the trait? `DA: DaService<Error = anyhow::Error> + Clone`

Reply:
done